e2e/pages/manage/v2MarkBedAsOutOfServicePage.ts

```diff
@@ -0,0 +1,67 @@
+import { Page, expect } from '@playwright/test'
+import { getDate, getMonth, getYear } from 'date-fns'
+import { faker } from '@faker-js/faker/locale/en_GB'
+import { BasePage } from '../basePage'
+
+export class V2MarkBedAsOutOfServicePage extends BasePage {
+  startDate: Date
+
+  static async initialize(page: Page, title?: string) {
+    if (title) {
+      await expect(page.locator('h1')).toContainText(title)
+    }
+    const instance = new V2MarkBedAsOutOfServicePage(page)
+    instance.startDate = faker.date.soon({ days: 600 })
+    return instance
+  }
+
+  endDate() {
+    const endDate = new Date(this.startDate)
+    endDate.setDate(endDate.getDate() + 2)
+    return endDate
+  }
+
+  async enterOutOfServiceFromDate() {
+    const { startDate } = this
+    await this.page
+      .getByRole('group', { name: 'Out of service from' })
+      .getByLabel('Day')
+      .fill(getDate(startDate).toString())
+    await this.page
+      .getByRole('group', { name: 'Out of service from' })
+      .getByLabel('Month')
+      .fill(getMonth(startDate).toString())
+    await this.page
+      .getByRole('group', { name: 'Out of service from' })
+      .getByLabel('Year')
+      .fill(getYear(startDate).toString())
+  }
+
+  async enterOutOfServiceToDate() {
+    const endDate = this.endDate()
+    await this.page
+      .getByRole('group', { name: 'Out of service to' })
+      .getByLabel('Day')
+      .fill(getDate(endDate).toString())
+    await this.page
+      .getByRole('group', { name: 'Out of service to' })
+      .getByLabel('Month')
+      .fill(getMonth(endDate).toString())
+    await this.page
+      .getByRole('group', { name: 'Out of service to' })
+      .getByLabel('Year')
+      .fill(getYear(endDate).toString())
```

Comment:
Not for this PR but I think it would be useful if we have a method on the `BasePage` class that fills a Date field input group given its name

Reply:
yes that will useful